addons/website_sale/controllers/main.py

```diff
@@ -441,6 +441,7 @@ def _prepare_product_values(self, product, category, search, **kwargs):
 
     @http.route(['/shop/change_pricelist/<model("product.pricelist"):pl_id>'], type='http', auth="public", website=True, sitemap=False)
     def pricelist_change(self, pl_id, **post):
+        redirect_url = None
         if (pl_id.selectable or pl_id == request.env.user.partner_id.property_product_pricelist) \
                 and request.website.is_pricelist_available(pl_id.id):
             redirect_url = request.httprequest.referrer
```

Comment:
It seems to me that it was introduced by accident, see [diff](https://github.com/odoo/odoo/pull/74575/files#diff-83686977d1607bdb0a3e755e9d22ec4496c1dc4caebfff4bee3902ddd1b3e866R406-R434)
I think we can always assign `redirect_url` to `request.httprequest.referrer`

```suggestion
        redirect_url = request.httprequest.referrer
        if (pl_id.selectable or pl_id == request.env.user.partner_id.property_product_pricelist) \
                and request.website.is_pricelist_available(pl_id.id):
```

Reply:
It also means we can combine both `if`, up to you.
```suggestion
        redirect_url = request.httprequest.referrer
        if (pl_id.selectable or pl_id == request.env.user.partner_id.property_product_pricelist) \
                and request.website.is_pricelist_available(pl_id.id) \
                and redirect_url and request.website.is_view_active('website_sale.filter_products_price'):
```

Reply:
> It seems to me that it was introduced by accident, see [diff](https://github.com/odoo/odoo/pull/74575/files#diff-83686977d1607bdb0a3e755e9d22ec4496c1dc4caebfff4bee3902ddd1b3e866R406-R434) I think we can always assign `redirect_url` to `request.httprequest.referrer`

Let's go with your first suggestion, as I think the second if is necessary for 
```python
            request.session['website_sale_current_pl'] = pl_id.id
            request.website.sale_get_order(force_pricelist=pl_id.id)
```

Reply:
Aah yes my bad 🙂 

Thanks